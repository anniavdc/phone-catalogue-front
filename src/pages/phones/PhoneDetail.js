import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

// files
import Loader from './../../components/loader/Loader';
import { phoneList } from './../../store/actions/index';

export default function PhoneDetail() {
    const phones = useSelector(state => state.phones.phoneList);
    const dispatch = useDispatch();
    let { phoneId } = useParams();

    useEffect(() => {
        // if there is no phone data
        if (!phones.length) {
            dispatch(phoneList());
        }
    }, [phones, dispatch]);

    const baseURL = process.env.REACT_APP_API_BASE;
    const selectedPhone = phones.find(phone => phone.id === parseInt(phoneId));
    return (
        !selectedPhone
            ? (
                <Loader extraClass="page" />
            ) : (
                <React.Fragment>
                    <h2 className="products-title">Phone Details</h2>
                    <div className="detail-wrapper">
                        <div className="detail-image" style={{ backgroundImage: `url(${baseURL}/public/${selectedPhone.imageFileName})` }}></div>
                        <div className="detail-content">
                            <h3 className="detail-title">{selectedPhone.name}</h3>
                            <p>{selectedPhone.manufacturer || ''}</p>
                            <p>{selectedPhone.description || ''}</p>
                            <div className="detail-features">
                                <p>Features</p>
                                <p>{selectedPhone.screen || ''}</p>
                                <p>{selectedPhone.processor || ''}</p>
                                <p>{selectedPhone.ram || ''}</p>
                                <p><span className="detail-color" style={{ backgroundColor: selectedPhone.color || 'white' }}></span>{selectedPhone.color || ''}</p>
                                <p className="detail-price">Price: {selectedPhone.price}</p>
                            </div>
                        </div>
                    </div>
                </React.Fragment>
            )
    )
}
