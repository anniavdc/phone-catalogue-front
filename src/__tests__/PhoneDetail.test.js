import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { waitForDomChange } from '@testing-library/dom';

// files
import render from '../utils/tests/renderWithRouter';
import PhoneDetail from '../pages/phones/PhoneDetail';

const initialState = {
    phones: {
        phoneList: [
            {
                color: "black",
                description: "This is iPhone 7.",
                id: 0,
                imageFileName: "IPhone_7.png",
                manufacturer: "Apple",
                name: "iPhone 7",
                price: 769,
                processor: "A10 Fusion",
                ram: 2,
                screen: "4,7 inch IPS"
            }
        ]
    }
};

const mockStore = configureStore([thunk]);

describe('PhoneDetail', () => {
    let store;
    beforeEach(() => {
        store = mockStore(initialState);
    })

    test('render', async () => {
        const { container } = render(
            <Provider store={store}>
                <PhoneDetail match={{ params: { phoneId: "0" }, isExact: true, path: "/app/:phoneId", url: "/app/0" }} />
            </Provider>,
            "/app/0",
            { wrapper: MemoryRouter }
        );
        expect(container).toMatchSnapshot();
    });

})
