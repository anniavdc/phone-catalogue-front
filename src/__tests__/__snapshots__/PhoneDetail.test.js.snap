// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetail render 1`] = `
<div>
  <div
    class="loader page"
  />
</div>
`;
